core/indexer/interface.go

```diff
@@ -2,44 +2,66 @@ package indexer
 	SetTxLogsProcessor(txLogsProc process.TransactionLogProcessorDatabase)
 	SaveBlock(body data.BodyHandler, header data.HeaderHandler, txPool map[string]data.TransactionHandler, signersIndexes []uint64, notarizedHeadersHashes []string)
-	SaveRoundsInfos(roundsInfos []RoundInfo)
+	RevertIndexedBlock(header data.HeaderHandler, body data.BodyHandler)
+	SaveRoundsInfo(roundsInfos []workItems.RoundInfo)
 	UpdateTPS(tpsBenchmark statistics.TPSBenchmark)
 	SaveValidatorsPubKeys(validatorsPubKeys map[uint32][][]byte, epoch uint32)
-	SaveValidatorsRating(indexID string, infoRating []ValidatorRatingInfo)
+	SaveValidatorsRating(indexID string, infoRating []workItems.ValidatorRatingInfo)
+	StopIndexing() error
 	IsInterfaceNil() bool
 	IsNilIndexer() bool
 }
 
-// DatabaseHandler is an interface used by elasticsearch component to prepare data to be saved on elasticsearch server
-type DatabaseHandler interface {
+// DispatcherHandler is an interface for the data dispatcher data is used to index data in database
+type DispatcherHandler interface {
+	StartIndexData()
+	Close() error
+	Add(item workItems.WorkItemHandler)
+}
+
+// ElasticProcessor defines the interface for the elastic search indexer
+type ElasticProcessor interface {
+	SaveShardStatistics(tpsBenchmark statistics.TPSBenchmark) error
+	SaveHeader(header data.HeaderHandler, signersIndexes []uint64, body *block.Body, notarizedHeadersHashes []string, txsSize int) error
+	RemoveHeader(header data.HeaderHandler) error
+	RemoveMiniblocks(header data.HeaderHandler, body *block.Body) error
+	SaveMiniblocks(header data.HeaderHandler, body *block.Body) (map[string]bool, error)
+	SaveTransactions(body *block.Body, header data.HeaderHandler, txPool map[string]data.TransactionHandler, selfShardID uint32, mbsInDb map[string]bool) error
+	SaveValidatorsRating(index string, validatorsRatingInfo []workItems.ValidatorRatingInfo) error
+	SaveRoundsInfo(infos []workItems.RoundInfo) error
+	SaveShardValidatorsPubKeys(shardID, epoch uint32, shardValidatorsPubKeys [][]byte) error
 	SetTxLogsProcessor(txLogsProc process.TransactionLogProcessorDatabase)
-	SaveHeader(header data.HeaderHandler, signersIndexes []uint64, body *block.Body, notarizedHeadersHashes []string, txsSize int)
-	SaveMiniblocks(header data.HeaderHandler, body *block.Body) map[string]bool
-	SaveTransactions(body *block.Body, header data.HeaderHandler, txPool map[string]data.TransactionHandler, selfShardID uint32, mbsInDb map[string]bool)
-	SaveRoundsInfos(infos []RoundInfo)
-	SaveShardValidatorsPubKeys(shardID, epoch uint32, shardValidatorsPubKeys [][]byte)
-	SaveValidatorsRating(Index string, validatorsRatingInfo []ValidatorRatingInfo)
-	SaveShardStatistics(tpsBenchmark statistics.TPSBenchmark)
 }
 
-// databaseClientHandler is an interface that do requests to elasticsearch server
-type databaseClientHandler interface {
+// DatabaseClientHandler is an interface that do requests to elasticsearch server
+type DatabaseClientHandler interface {
```

Comment:
IsInterfaceNil() method is not added here.

Reply:
done.